raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -958,7 +959,10 @@ private CompletableFuture<FetchResponseData> handleFetchRequest(
                 Errors.INVALID_REQUEST, Optional.empty()));
         }
 
-        FetchResponseData response = tryCompleteFetchRequest(request.replicaId(), fetchPartition, currentTimeMs);
+        FetchResponseData response = tryCompleteFetchRequest(
+            FetchRequest.replicaId(request),
```

Comment:
There are other usages of `request.replicaId()` is this file. Could we extend unit tests to cover them all?

Reply:
Done.